src/dynamic-route-resolver/views/DynamicRouteResolver.tsx

```diff
@@ -331,6 +368,27 @@ const DynamicRouteResolver: FC<DynamicRouteResolverProps> = ({
 				</ErrorView>
 			);
 		}
+		if (routeInfo && routeInfo.type === 'pupilOnlyPage') {
+			return (
+				<ErrorView icon={IconName.clock} actionButtons={['help', 'helpdesk']} message="">
+					{GET_ERROR_MESSAGES()[`PUPIL_ONLY`]}
+				</ErrorView>
+			);
+		}
+		if (routeInfo && routeInfo.type === 'notForPupilPage') {
+			return (
+				<ErrorView icon={IconName.clock} actionButtons={['pupils']} message="">
+					{GET_ERROR_MESSAGES()[`NOT_FOR_PUPILS`]}
+				</ErrorView>
+			);
+		}
+		if (routeInfo && routeInfo.type === 'wrongUserGroupPage') {
+			return (
+				<ErrorView icon={IconName.clock} actionButtons={['help', 'helpdesk']} message="">
+					{GET_ERROR_MESSAGES()[`OTHER_ROLES`]}
+				</ErrorView>
+			);
```

Comment:
message is not required
so you can either drop it, or you could pass the error message through that way instead of the children. either is fine.